### PR TITLE
UpStreamCheckUtils detection failing when upstream using hostname without port

### LINKED ISSUE #1235: UpstreamCheckUtils检测不通过
## Describe the bug
当插件selector的upstream使用主机名时, UpstreamCheckUtils检测不通过

## Environment

- Soul version(s): [e.g. 2.3.0-RELEASE]

## Expected behavior
1. 修复当插件selector的upstream使用主机名时, UpstreamCheckUtils检测不通过
2. upstreamUrl没有端口信息时，默认检测80端口

## Debug logs

<details>
<summary>Click to expand</summary>

```
replace this line with your debug logs
```
</details>

### PR DESCRIPTION
**Pull request by ywengineer**
// Describe your PR here; eg. Fixes #1235 

Make sure that:

- [x] You have read the [contribution guidelines](https://dromara.org/projects/soul/contributor/).
- [x] You submit test cases (unit or integration tests) that back your changes.
- [x] Your local test passed `mvn clean install -Dmaven.javadoc.skip=true`.

### CODE REVIEW

**Comment by ywengineer** [comment]
@yu199195

**Comment by yu199195** [comment]
@ywengineer  can you modify title to english ?

File: soul-common/src/main/java/org/dromara/soul/common/utils/UpstreamCheckUtils.java [hunk 1]
```diff
@@ -47,17 +47,18 @@ public static boolean checkUrl(final String url) {
...
-            return isHostConnector(hostPort[0], Integer.parseInt(hostPort[1]));
+        String[] hostPort;
+        if (url.startsWith(HTTP)) {
+            final String[] http = StringUtils.split(url, "\\/\\/");
+            hostPort = StringUtils.split(http[1], Constants.COLONS);
         } else {
-            return isHostReachable(url);
+            hostPort = StringUtils.split(url, Constants.COLONS);
+        }
+        final int port = hostPort.length > 1 ? Integer.parseInt(hostPort[1]) : 80;
```
Comment (dengliming) [inline soul-common/src/main/java/org/dromara/soul/common/utils/UpstreamCheckUtils.java:57]: It it possible to support ` https(443)`?

Comment (ywengineer) [reply soul-common/src/main/java/org/dromara/soul/common/utils/UpstreamCheckUtils.java]: done

**Comment by yu199195** [comment]
please check ci passed

**Comment by ywengineer** [comment]
Failed to execute goal on project soul-metrics-prometheus: Could not resolve dependencies for project org.dromara:soul-metrics-prometheus:jar:2.3.1-SNAPSHOT: The following artifacts could not be resolved: org.dromara:soul-metrics-spi:jar:2.3.1-SNAPSHOT, org.dromara:soul-common:jar:2.3.1-SNAPSHOT: Could not find artifact org.dromara:soul-metrics-spi:jar:2.3.1-SNAPSHOT 

this is not an issue.

**Pull request merged**